### PR TITLE
Updating Parse.LiveQuery to use currentUserAsync.

### PR DESCRIPTION
**Pull request by peterdotjs**
Addressing #232.

### CODE REVIEW

File: src/ParseLiveQuery.js [hunk 1]
```diff
@@ -69,16 +70,19 @@ LiveQuery.on('error', () => {
 export default LiveQuery;
 
 let getSessionToken = () => {
-  let currentUser = CoreManager.getUserController().currentUser();
-  let sessionToken;
-  if (currentUser) {
-    sessionToken = currentUser.getSessionToken();
-  }
-  return sessionToken;
+  let promiseUser = CoreManager.getUserController().currentUserAsync();
```
Comment (wangmengyan95) [inline src/ParseLiveQuery.js]: It is wired to create promiseXXX for each inner promises, why not just chain them?

File: src/ParseLiveQuery.js [hunk 2]
```diff
@@ -86,65 +90,76 @@ let DefaultLiveQueryController = {
...
+
+      return ParsePromise.as(defaultLiveQueryClient);
     });
-    return defaultLiveQueryClient;
   },
   open() {
-    let liveQueryClient = getLiveQueryClient();
-    liveQueryClient.open();
+    getLiveQueryClient().then((liveQueryClient) => {
+      return ParsePromise.as(liveQueryClient.open());
```
Comment (wangmengyan95) [inline src/ParseLiveQuery.js]: Discuss offline, we do not need to wrap it around a new promise.

**Comment by joshblour** [comment]
I tried this out in my react-native app. `query.subscribe()` now returns a promise and the following example fails:
```js
		let subscription = query.subscribe()
		subscription.on('open', () => {
 			console.log('subscription opened');
		});
```

**Comment by peterdotjs** [comment]
I'm working on changing this to keep the interface the same - when landed .subscribe will return a subscription.

File: src/LiveQuerySubscription.js [hunk 3]
```diff
@@ -102,8 +102,11 @@ export default class Subscription extends EventEmitter {
    * @method unsubscribe
    */
   unsubscribe() {
-    var liveQueryClient = CoreManager.getLiveQueryController().getDefaultLiveQueryClient();
-    liveQueryClient.unsubscribe(this);
-    this.emit('close');
+    let _this = this;
```
Comment (wangmengyan95) [inline src/LiveQuerySubscription.js]: Do we need this? I think the arrow function auto bind this for you?

File: src/ParseLiveQuery.js [hunk 4]
```diff
@@ -86,65 +89,103 @@ let DefaultLiveQueryController = {
...
+      // Get currentUser sessionToken if possible
+      defaultLiveQueryClient = new LiveQueryClient({
+        applicationId,
+        serverURL: liveQueryServerURL,
+        javascriptKey,
+        masterKey,
+        sessionToken,
+      });
+      // Register a default onError callback to make sure we do not crash on error
+      defaultLiveQueryClient.on('error', (error) => {
```
Comment (wangmengyan95) [inline src/ParseLiveQuery.js]: I do not know we can chain the `on` method, let's clean this.

File: src/ParseLiveQuery.js (hunk 1 above)
Comment (wangmengyan95) [inline src/ParseLiveQuery.js]: Any reason you want to keep this?

File: src/LiveQuerySubscription.js (hunk 3 above)
Comment (peterdotjs) [inline src/LiveQuerySubscription.js]: arrow does auto bind but in this case we'll need to distinguish between the promise and the subscription context.

File: src/ParseLiveQuery.js (hunk 1 above)
Comment (peterdotjs) [inline src/ParseLiveQuery.js]: what is the issue here? I thought it's more readable than chaining directly from the above.

File: src/ParseLiveQuery.js (hunk 4 above)
Comment (peterdotjs) [inline src/ParseLiveQuery.js]: sure ill update this

**Pull request merged**